src/audio/pipeline/pipeline-graph.c

```diff
@@ -348,6 +349,9 @@ static int pipeline_comp_reset(struct comp_dev *current,
 	/* reset should propagate to the connected pipelines,
 	 * which need to be scheduled together
 	 */
+	if (!is_single_ppl && IPC4_MOD_ID(current->ipc_config.id))
+		return 0;
```

Comment:
Maybe explain in the commit what's the logic behind the "IPC4_MOD_ID()" check. Without context, this seems hard to understand.

Reply:
Updated